browser/speedreader/page_distiller.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+#ifndef BRAVE_BROWSER_SPEEDREADER_PAGE_DISTILLER_H_
+#define BRAVE_BROWSER_SPEEDREADER_PAGE_DISTILLER_H_
+
+#include <string>
+
+#include "base/functional/callback_forward.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
+#include "base/observer_list_types.h"
+#include "base/values.h"
+#include "brave/components/speedreader/speedreader_util.h"
+
+namespace content {
+class WebContents;
+}
+
+namespace speedreader {
+
+class PageDistiller {
+ public:
+  enum class State {
+    kUnknown,
+    kNotDistillable,
+    kDistillable,
+    kDistilled,
+  };
+
+  class Observer : public base::CheckedObserver {
+   public:
+    virtual void OnPageDistillStateChanged(State state) {}
+
+   protected:
+    ~Observer() override = default;
+  };
+
+  using DistillContentCallback =
+      base::OnceCallback<void(bool success, std::string content)>;
+
+  State GetState() const;
+
+  void AddObserver(Observer* observer);
+  void RemoveObserver(Observer* observer);
+
+  void GetDistilledHTML(DistillContentCallback callback);
+  void GetDistilledText(DistillContentCallback callback);
+
+ protected:
+  explicit PageDistiller(content::WebContents* web_contents);
+  virtual ~PageDistiller();
+
+  void SetWebContents(content::WebContents* web_contents);
```

Comment:
is it used for anything else then setting `null`? If yes, this is either `ResetWebContents()` or `WebContentsDestroyed()`